bin/plugin/commands/build-plugins.js

```diff
@@ -0,0 +1,200 @@
+					updateModuleDetails( {
+						pluginPath: buildModulePath,
+						regex: '@package\\s{1,}performance-lab',
+						result: '@package ' + pluginSlug,
+					} );
+				} catch ( error ) {
+					log(
+						formats.error(
+							`${ error }`
+						)
+					);
+				}
+			}
+		} catch ( jsonError ) {
+			log(
+				formats.error(
+					`Error parsing JSON string: "${ jsonError }"`
+				)
+			);
+		}
+	} );
+};
+
+/**
+ * Updates the `load.php` file header information.
+ *
+ * @param {[]Settings} settings Plugin settings.
+ */
+async function updatePluginHeader( settings ) {
+	const { key, version, slug, pluginPath } = settings;
+	// Plugin root `load.php` file content.
+	const rootLoadFile = path.join( '.', 'load.php' );
+	const rootLoadFileContent = fs.readFileSync( rootLoadFile, 'utf-8' );
+	// Specific module `load.php` file content.
+	const buildLoadFile = path.join( pluginPath, 'load.php' );
+	const buildLoadFileContent = fs.readFileSync( buildLoadFile, 'utf-8' );
+
+	const rootFileHeader = await getModuleHeader( rootLoadFileContent );
+	const buildFileLoad = await getModuleHeader( buildLoadFileContent );
+
+	// Get module header data.
+	const { name, description } = await getModuleData( buildLoadFileContent );
+
+	// Replace plugin root's `load.php` header in the module file header.
+	fs.writeFileSync( buildLoadFile, buildLoadFileContent.replace( buildFileLoad, rootFileHeader ) );
+
+	let updatedBuildLoadFileContent = fs.readFileSync( buildLoadFile, 'utf-8' );
+
+	// Get module header data.
+	const moduleData = {
+		pluginname: name,
+		plugindescription: description,
+		pluginuri: '/tree/trunk/modules/' + key,
+		pluginversion: version,
+		textdomain: slug,
+	};
+
+	// Map of module header => object property.
+	const headers = {
+		'Plugin Name': 'pluginname',
+		'Plugin URI': 'pluginuri',
+		Description: 'plugindescription',
+		Version: 'pluginversion',
+		'Text Domain': 'textdomain',
+	};
+
+	const headerRegex = new RegExp( `(?<header>${ Object.keys( headers ).join( '|' ) }): (?<value>(.*?)+)`, 'gmi' );
+	const matches = [ ...updatedBuildLoadFileContent.matchAll( headerRegex ) ];
+
+	matches.forEach( ( match ) => {
+		const { header, value } = match.groups;
+		const prop = headers[ header ];
+		let moduleDataValue = moduleData[ prop ];
+
+		if ( value && prop && moduleDataValue ) {
+			if ( prop === 'pluginuri' ) {
+				moduleDataValue = value + moduleDataValue;
+			}
+			fs.writeFileSync( buildLoadFile, updatedBuildLoadFileContent.replace( new RegExp( value ), moduleDataValue ) );
+			updatedBuildLoadFileContent = fs.readFileSync( buildLoadFile, 'utf-8' );
+		}
+	} );
+}
+
+/**
+ * Gets the specific module file header information.
+ *
+ * @param {string} moduleFileContent File content.
+ *
+ * @return {[]HeaderData} File header data.
+ */
+async function getModuleData( moduleFileContent ) {
+	const moduleHeader = await getModuleHeader( moduleFileContent );
+	return await getModuleDataFromHeader( moduleHeader );
+}
```

Comment:
I don't think we need this function. See my related comment above, we already get the module header above, and we need it separately to replace it later (see also https://github.com/WordPress/performance/pull/662#discussion_r1128192056). So we should remove this function and instead call the two individual functions in here directly.